cypress/tests/upstream/vm-adv-search.cy.ts

```diff
@@ -0,0 +1,227 @@
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('not.exist');
+      cy.get(ADV_SRCH_BTN).click();
+      cy.get('[data-test="adv-search-vm-labels-toggle"]').find('input').type('u1');
+      cy.byButtonText('instancetype/u1.small=true').click();
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('exist');
+    });
+
+    it('test VM search by vCPU', () => {
+      cy.get(ADV_SRCH_BTN).click();
+      cy.get('input[data-test="adv-search-vcpu-value"]').type('1');
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('not.exist');
+      cy.get(ADV_SRCH_BTN).click();
+      cy.get('button[data-test="adv-search-vcpu-operator-toggle"]').click();
+      cy.byButtonText('Less than').click();
+      cy.get('input[data-test="adv-search-vcpu-value"]').type('2');
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('exist');
+    });
+
+    it('test VM search by memory', () => {
+      cy.get(ADV_SRCH_BTN).click();
+      cy.get('input[data-test="adv-search-mem-value"]').type('2');
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('not.exist');
+      cy.get(ADV_SRCH_BTN).click();
+      cy.get('button[data-test="adv-search-mem-operator-toggle"]').click();
+      cy.byButtonText('Less than').click();
+      cy.get('input[data-test="adv-search-mem-value"]').type('4');
+      cy.clickSearchBtn();
+      cy.contains(NAME_CELL, 'centos-stream9-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'centos-stream10-sssss').should('not.exist');
+      cy.contains(NAME_CELL, 'fedora-sssss').should('exist');
+    });
+
+    it('test VM search by IP', () => {
+      SRCH_VMS.forEach((name) => {
+        cy.exec(`oc get vmi ${name} -n ${SEARCH_NS} -o yaml | grep ipAddress:`, {
+          failOnNonZeroExit: false,
+        })
+          .its('stdout')
+          .then((stdout) => {
+            const ip = stdout.split(': ')[1];
+            if (ip) {
+              cy.get(ADV_SRCH_BTN).click();
+              cy.get('.pf-v6-c-modal-box[role="dialog"]').within(() => {
+                cy.byButtonText('Network').click();
+                cy.get('input[data-test="adv-search-vm-ip"]').type(ip);
```

Comment:
here too avoid finding by text/classname and use data-test-id

Reply:
it does not exist in modals